### PR TITLE
Refactor tts function

### PR DESCRIPTION
**Pull request by oharan2**
Current implement will try to return an int when it gets invalid phrase like "H1"

### CODE REVIEW

**Review by myakove — COMMENTED** [review]
In the current implementation we do not raise unless the original value is not int. (Let int() do the raise if needed)

We do not support 1.5H, if the user want 1.5H the user should send 90m

**Comment by oharan2** [comment]
> In the current implementation we do not raise unless the original value is not int. (Let int() do the raise if needed)
> 
> We do not support 1.5H, if the user want 1.5H the user should send 90m

Done
Updated with just making it more strict towards the input
PR description is updated

**Review by myakove — CHANGES_REQUESTED** [review]
Let's keep it simple, we only accept:
`(number)/s/m/h` or `int` 
No need to get all permutation, the user should know how to use this function (docstring)

**Review by omrirh — APPROVED** [review]

File: app/libs/rosa_clusters.py [hunk 1]
```diff
@@ -33,20 +33,19 @@ def tts(ts):
...
-    _unit = time_and_unit["unit"].lower()
-    if _unit == "s":
-        return _time
-    elif _unit == "m":
-        return _time * 60
-    elif _unit == "h":
-        return _time * 60 * 60
-    else:
-        return int(ts)
+        print(
```
Comment (myakove) [inline app/libs/rosa_clusters.py:40]: You need to return `int(ts)` here.

File: app/libs/rosa_clusters.py [hunk 2]
```diff
@@ -33,20 +33,19 @@ def tts(ts):
...
+            f"Time format is not valid. Expecting s/m/h time value or int, got '{ts}'."
+        )
+        raise
+
+    time_ref = {"s": 1, "m": 60, "h": 60 * 60}
+    if _unit:
+        return int(_val) * time_ref[_unit]
+    return int(_val)
```
Comment (myakove) [inline app/libs/rosa_clusters.py:48]: just `return int(_val) * time_ref[_unit]`

**Review by myakove — CHANGES_REQUESTED** [review]

**Comment by oharan2** [comment]
> You need to return `int(ts)` here.

If we will keep the regex 'strictness' and the exception handling in the same way as before,
I'm closing since i'm not seeing added value in my changes.

**Pull request closed without merging**